parallel-consumer-reactor/src/main/java/io/confluent/parallelconsumer/reactor/ReactorProcessor.java

```diff
@@ -102,26 +103,10 @@ public void react(Function<PollContext<K, V>, Publisher<?>> reactorFunction) {
                     // it will run in the controller thread, unless user themselves uses either publishOn or successful
                     // subscribeOn
                     .publishOn(getScheduler())
-                    .doOnNext(signal -> {
-                        log.trace("doOnNext {}", signal);
-                    })
-                    .doOnComplete(() -> {
-                        log.debug("Reactor success (doOnComplete)");
-                        pollContext.streamWorkContainers().forEach(wc -> {
-                            wc.onUserFunctionSuccess();
-                            addToMailbox(pollContext, wc);
-                        });
-                    })
-                    .doOnError(throwable -> {
-                        log.error("Reactor fail signal", throwable);
-                        pollContext.streamWorkContainers().forEach(wc -> {
-                            wc.onUserFunctionFailure(throwable);
-                            addToMailbox(pollContext, wc);
-                        });
-                    })
+                    .doOnNext(signal -> log.trace("doOnNext {}", signal))
                     // cause users Publisher to run a thread pool, if it hasn't already - this is a crucial magical part
                     .subscribeOn(getScheduler())
-                    .subscribe();
+                    .subscribe(ignore -> onComplete(pollContext), throwable -> onError(pollContext, throwable));
```

Comment:
Hey @brp-jakob - can you please explain why the change from `doOnComplete(successCallback)` / `doOnError(errorCallback)` is needed to `subscribe(successCallback, errorCallback)` ?

Reply:
Its this part:
"Error handling moved to subscribe param to avoid onErrorDropped."

There is a long discussion about this here (and in related issues to it): https://github.com/reactor/reactor-core/issues/1431